File: back-end/src/main/java/com/trucks/application/port/BerthRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.trucks.domain.entity.Berth;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface BerthRepository {

    Berth save(Berth berth);

    Optional<Berth> find(UUID id);

    Page<Berth> findAll(PageRequest pageRequest);

    Berth update(UUID id, Berth berth);
    
    Berth delete(UUID id);

    List<Berth> deleteAll();
}

File: front-end/src/application/repository/berthRepository.ts
import { Berth } from "../../domain/entity/berth";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";

export interface BerthRepository {
    save: (berth: Berth) => Promise<Either<DataError, Berth>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Berth[]>>;
    
    deleteAll: () => Promise<Either<DataError, Berth[]>>;
}